cdap-spark-core/src/main/java/co/cask/cdap/app/runtime/spark/SparkProgramRuntimeProvider.java

```diff
@@ -57,23 +57,18 @@ public ProgramRunner createProgramRunner(ProgramType type, Mode mode, Injector i
                                 type, ProgramType.SPARK);
     switch (mode) {
       case LOCAL:
-        return createSparkProgramRunner(injector);
+        return createSparkProgramRunner(injector, PROGRAM_RUNNER_CLASS_NAME);
       case DISTRIBUTED:
-        return injector.getInstance(DistributedSparkProgramRunner.class);
+        return createSparkProgramRunner(injector, DISTRIBUTED_PROGRAM_RUNNER_CLASS_NAME);
```

Comment:
This is to make sure we don't have classloader leak in distributed mode as well.